app/controllers/tags_controller.rb

```diff
@@ -63,6 +63,55 @@ def destroy
     end
   end
 
+  def paginated_posts
+    @tag = Tag.find params[:id]
+    if @tag.posts
```

Comment:
i think this will return a 500 if there are no `@tag.posts`, so this should return something using `respond_to format` json, ideally with a helpful error message like "No posts found".